lib/permission.js

```diff
@@ -94,16 +92,19 @@ SELECT DISTINCT "Application"."Name" FROM "Application"
     SELECT "Roles" FROM "SystemUserRoles" WHERE "SystemUser" = $1
   )`;
 
-const checkPermission = (provider, accessType, category, userId, callback) => {
+const checkPermission = async (provider, accessType, category, userId) => {
   const flag = permissionFlags[accessType];
-  provider.pool.query(categoryQuery(), [userId, category, flag], (err, res) => {
-    if (err) {
-      provider.systemLogger(err);
-      callback(err);
-    } else {
-      callback(null, res.rows[0].exists);
-    }
-  });
+  try {
+    const res = await provider.pool.query(categoryQuery(), [
```

Comment:
```js
    const res = await provider.pool.query(
      categoryQuery(),
      [userId, category, flag]
    );
```
?

Reply:
I don't see the difference, is it just formatting? I didn't format the code manually, it is done automatically by Prettier.

Reply:
Oh, if that was Prettier then this is fine.